src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -207,7 +207,19 @@ protected Object readResolve() {
     }
 
     public void taskAccepted(Executor executor, Queue.Task task) {
-        return;
+        EC2Computer computer = (EC2Computer) executor.getOwner();
+        EC2AbstractSlave slaveNode = computer.getNode();
+        int maxTotalUses = slaveNode.maxTotalUses;
+        if (maxTotalUses <= -1) {
+            LOGGER.info("maxTotalUses set to unlimited (" + slaveNode.maxTotalUses + ") for agent " + slaveNode.instanceId);
+            return;
+        } else if (maxTotalUses <= 1) {
+            LOGGER.info("maxTotalUses drained - suspending agent " + slaveNode.instanceId);
+            computer.setAcceptingTasks(false);
+        } else {
+            slaveNode.maxTotalUses = slaveNode.maxTotalUses - 1;
+            LOGGER.info("maxTotalUses set to " + slaveNode.maxTotalUses + " for agent " + slaveNode.instanceId);
```

Comment:
This message is slight misleading in my opinion. Instead of `maxTotalUses set to`, maybe something like `slaveNode.instanceId + "  has " + slaveNode.maxTotalUses + " left"`